src/main/kotlin/com/exactpro/th2/rptdataprovider/handlers/IParentEventCounter.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.rptdataprovider.handlers
+
+import com.exactpro.th2.rptdataprovider.entities.responses.BaseEventEntity
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+
+internal interface IParentEventCounter {
+    /**
+     * This method use parent event id or event id to limit number of child events.
+     * WARNING: event id isn't grantee event unique then this method can't be used for strict limitation.
+     * @return false if limit exceeded otherwise true
+     */
+    fun checkCountAndGet(event: BaseEventEntity): Boolean
+
+    private class NoLimitedParentEventCounter : IParentEventCounter {
```

Comment:
suggestion: you can make it an `object` instead of class. This way the app will have only one instance of this object (without having a constant definition)